extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/PentahoMetadataDomainRepository.java

```diff
@@ -645,7 +785,7 @@ private void internalReloadDomains() {
       for ( final RepositoryFile child : children ) {
         if ( getAclHelper().canAccess( child, READ ) ) {
           // Get the metadata for this file
-          final Map<String, Serializable> fileMetadata = repository.getFileMetadata( child.getId() );
+          final Map<String, Serializable> fileMetadata = getFileMetadataHelper( child.getId() );
```

Comment:
A domain is considered misgrated once it has the file metadata property “datasource-type”. The logic for migrateDomain permanently adds this property “data source-type”. once a domain is migrated, this property will persist between launches of the system.

When a domain is migrated, the code wont add the same property twice.

So after a domain is migrated the flag, system.migrate-domain has no effect.

Reply:
There’s two things our are touching on. `InternalReloadDomains` just loads walks the JCR tree an crates a map with the key being domain identifier and the value being the JCR Path which is hash code. 

`InternalReloadDomains` Has always been a single threaded. BACKLOG-35854 threads the data-access services that call `getDomain` https://github.com/pentaho/data-access/commit/84bc9c26648afc634a6bddbf324db19e20cffbce

Reply:
I always lean towards allowing things to be configurable. In the unknown scenario, someone would want to turn off migration, they can. Engineering, doesn't have to make a new jar just for this case.

Reply:
In my original comment, I wrote:

> When the configuration "system.migrate-domains" is false, the domain will not be indexed against data source type and will later fail to be retrieved by either getDataSourceWizardDomainIds or getMetadataDomainIds. Dependent functionality will, effectively break. Why would we want to support such a configuration? I thought the config was for avoiding updating the repository with the new metadata, yet, when false, we'd still need to be able to properly answer those methods (and only have the data source type metadata in memory).

Why would an option be supported that allows the code to malfunction?
Is my interpretation incorrect, that if the flag is initially`false`, before conversion has taken place, code will malfunction?

Reply:
By setting to false it will work that way it has been working in previous release.

Reply:
Lets provide a comment as why someone will set the value to false

Reply:
@rmansoor 
> By setting to false it will work that way it has been working in previous release.

Per my understanding of the code, that's not the case. If the flag is `false`, indexing will not take place and `getDataSourceWizardDomainIds` or `getMetadataDomainIds` will return empty collections.

@NJtwentyone am I wrong?

Reply:
"Why would an option be supported that allows the code to malfunction?"

I lead towards always having things configurable. It helps out end users, or developers. I lean towards not limiting functionality. Maybe someone(user/dev/customer service rep) is troubleshooting something and wants to make sure the "migration" isn't happening. The user can always "shoot themselves in the foot" with all configuration.

However, on a new installation, the system.property file will have that flag set to true. 
If the system.property file has a the missing property or it's malformed, the flag will be set to true.

Reply:
@NJtwentyone 

I don't understand why you keep not answering [my question](https://github.com/pentaho/pentaho-platform/pull/5124#discussion_r957754499).

Reply:
Resolving this, given removal of `canMigrate`, https://github.com/pentaho/pentaho-platform/pull/5124/commits/31008cd137fe8ee97aa521dc078372beac5de06b.